broker/src/bin/neon_broker.rs

```diff
@@ -0,0 +1,563 @@
+                NUM_SUBS_ALL.set(self.subs_to_all.len() as i64);
+                SubAnnounce::AddAll(sub_tx)
+            }
+            SubscriptionKey::Timeline(ttid) => {
+                self.subs_to_timelines
+                    .entry(ttid)
+                    .or_default()
+                    .push(SubSender(sub_id, sub_tx.clone()));
+                self.num_subs_to_timelines += 1;
+                NUM_SUBS_TIMELINE.set(self.num_subs_to_timelines);
+                SubAnnounce::AddTimeline(ttid, SubSender(sub_id, sub_tx))
+            }
+        };
+        // Collect existing publishers to notify them after lock is released;
+        // TODO: the probability of channels being full here is tiny (publisher
+        // always blocks listening chan), we can try sending first and resort to
+        // cloning if needed.
+        //
+        // Deadlock is possible only if publisher tries to access shared state
+        // during its lifetime, i.e. we add maintenance of set of published
+        // tlis. Otherwise we can just await here (but lock must be replaced
+        // with Tokio one).
+        //
+        // We could also just error out if some chan is full, but that needs
+        // cleanup of incompleted job, and notifying publishers when unregistering
+        // is mandatory anyway.
+        (sub_id, self.pub_txs.values().cloned().collect(), announce)
+    }
+
+    // Unregister the subscriber. Similar to register_subscriber, returns list
+    // of channels through which publishers must be notified about the removal.
+    pub fn unregister_subscriber(
+        &mut self,
+        sub_id: SubId,
+        sub_key: SubscriptionKey,
+    ) -> Option<(Vec<Sender<SubAnnounce>>, SubAnnounce)> {
+        // We need to notify existing publishers only about per timeline
+        // subscriptions, 'all' kind is detected on its own through closed
+        // channels.
+        let announce = match sub_key {
+            SubscriptionKey::All => {
+                assert!(self.subs_to_all.contains_key(&sub_id));
+                self.subs_to_all.remove(&sub_id);
+                NUM_SUBS_ALL.set(self.subs_to_all.len() as i64);
+                None
+            }
+            SubscriptionKey::Timeline(ref ttid) => {
+                remove_sub(&mut self.subs_to_timelines, ttid, sub_id);
+                self.num_subs_to_timelines -= 1;
+                NUM_SUBS_TIMELINE.set(self.num_subs_to_timelines);
+                Some(SubAnnounce::RemoveTimeline(*ttid, sub_id))
+            }
+        };
+        announce.map(|a| (self.pub_txs.values().cloned().collect(), a))
+    }
+}
+
+// SharedState wrapper for post-locking operations (sending to pub_tx chans).
+#[derive(Clone)]
+struct Registry {
+    shared_state: Arc<Mutex<SharedState>>,
+}
+
+impl Registry {
+    // Register new publisher in shared state.
+    pub fn register_publisher(&self) -> Publisher {
+        let (announce_tx, announce_rx) = mpsc::channel(128);
```

Comment:
> 128

Could that be configured too? Trying to understand what does it limit and can we somehow hit that limit during our regular work, as with the 256 one above.